src/role.rs

```diff
@@ -47,15 +48,13 @@ pub enum BlockchainRole {
 /// An arbitrating is the blockchain which will act as the decision engine, the arbitrating
 /// blockchain will use transaction to transfer the funds on both blockchains.
 pub trait Arbitrating:
-    Blockchain + Keys + Commitment + Signatures + Curve + Script + Onchain + Fee
+    Blockchain + Keys + Commitment + Signatures + Curve + Script + Onchain + Fee + Clone
 {
-    /// Defines the address format for the arbitrating blockchain
-    type Address;
-
-    //// Defines the type of timelock used for the arbitrating transactions
+    /// Defines the type of timelock used for the arbitrating transactions
     type Timelock: Copy;
+    /// Defines the address format for the arbitrating blockchain
+    type Address: StrictEncode + StrictDecode;
 }
-
 /// An accordant is the blockchain which does not need transaction inside the protocol nor
 /// timelocks, it is the blockchain with the less requirements for an atomic swap.
-pub trait Accordant: Blockchain + Keys + Curve + Commitment {}
+pub trait Accordant: Blockchain + Keys + Curve + Commitment + Clone + PrivateViewKey {}
```

Comment:
`PrivateViewKey` I don't like this as it is too specific to Monero. This should be fixed latter.

Reply:
view is a field of the generic `RevealAliceSessionParams` protocol message, could be something like accessory data
``` rust
    /// The `K_s^a` spend public key
    pub view: Ac::PrivateViewKey,
```

Reply:
Yes, this is already non-generic. I propose we keep it as it is now and we do latter a PR that apply genericness everywhere.